src/workerd/jsg/buffersource.h

```diff
@@ -102,9 +102,10 @@ class BackingStore {
 
   // Creates a new BackingStore of the given size.
   template <BufferSourceType T = v8::Uint8Array>
-  static BackingStore alloc(Lock& js, size_t size) {
-    return BackingStore(js.allocBackingStore(size), size, 0, getBufferSourceElementSize<T>(),
-        construct<T>, checkIsIntegerType<T>());
+  static BackingStore alloc(
+      Lock& js, size_t size, Lock::AllocOption init_mode = Lock::AllocOption::ZERO_INITIALIZED) {
```

Comment:
I'm really not a big fan of adding the `init_mode` option here. My preference would be to stay on the safer side and always have these allocations be zero-initialized as it makes it significantly less likely that we'll mess up and introduce a critical memory vuln. If we add this we need to be extra careful about how we use this.

Reply:
Really just a preference. We've been bitten before by uninitialized memory and even leaking data from one request to a next can be problematic. Still generally not a fan but consider it non-blocking.